### DISCUSSION TITLE
Prevent Prettier/ESLint from breaking long Tailwind html classes into multiple lines?

### DISCUSSION THREAD

**Discussion by diakonovm** [tailwindlabs/tailwindcss#4411] [Help]
I want to prevent Prettier/ESLint from breaking long Tailwind CSS class strings into multiple lines, as shown:

```
<span
  class="
    inline-flex
    items-center
    px-3
    rounded-l-md
    border border-r-0 border-gray-300
    bg-gray-50
    text-gray-500
    sm:text-sm
  "
>
```

I have the following configuration for Prettier and ESLint:
```
{
  "trailingComma": "none",
  "tabWidth": 2,
  "semi": false,
  "singleQuote": true,
  "quoteProps": "consistent",
  "printWidth": 100
}

```
```
module.exports = {
  root: true,
  env: {
    browser: true,
    node: true
  },
  parserOptions: {
    parser: 'babel-eslint'
  },
  extends: ['@nuxtjs', 'plugin:prettier/recommended', 'plugin:nuxt/recommended'],
  plugins: [],
  // add your custom rules here
  rules: {
    'vue/max-len': [
      'error',
      {
        code: 100,
        template: 100,
        tabWidth: 2,
        ignoreUrls: true,
        ignoreStrings: true,
        ignoreComments: true,
        ignoreHTMLAttributeValues: true
      }
    ],

    'vue/no-v-html': 'off'
  }
}

```

I generally like the prettier configuration, but would like to prevent this specific behavior for the class attribute. Is it possible to add an exception to prettier for the `printWidth` option only for `class` attributes?

**Comment by samarmohan** [comment]
This isn't a `tailwindcss` specific problem, it is ESLint and Prettier's job to make the code clean and easier to read, and IMO it accomplishes the job.

**Comment by simonswiss** [comment]
Hey!

Unfortunately I _don't think_ that there is a way to opt out of this formatting in Prettier version `2.3`. As someone recording screencasts on Tailwind CSS, this new feature is at times quite problematic, and what I've done for now is stick to version `2.2`.

At times I find that feature useful for educational purposes - isolating all the `border` utilities on the same line is _nice_ when doing a video on border utilities - but in many other situations, it definitely makes Tailwind flavoured markup excessively vertical 😅

Here's a link to the merged PR thread, if you want further context:

https://github.com/prettier/prettier/pull/7865
